support-data-rest/src/main/java/uk/ac/ebi/uniprot/api/support_data/controller/SuggesterController.java

```diff
@@ -33,8 +37,47 @@ public SuggesterController(SuggesterService suggesterService) {
     public ResponseEntity<Suggestions> suggester(
             @RequestParam(value = "dict", required = true) String dict,
             @RequestParam(value = "query", required = true) String query) {
+
+        return new ResponseEntity<>(suggesterService.findSuggestions(dict, query),
+                                    HttpStatus.OK);
+    }
+
+    @ExceptionHandler(MissingServletRequestParameterException.class)
+    public ResponseEntity<Error> handleMissingParams(MissingServletRequestParameterException ex) {
+        HttpStatus httpStatus = HttpStatus.BAD_REQUEST;
+        return new ResponseEntity<>(
+                Error.builder()
+                        .message("Missing required parameter: " + ex.getParameterName())
+                        .code(httpStatus.value())
+                        .build(), httpStatus);
+    }
+
+    @ExceptionHandler({UnknownDictionaryException.class})
+    public ResponseEntity<Error> handleMissingParams(UnknownDictionaryException ex) {
+        HttpStatus httpStatus = HttpStatus.BAD_REQUEST;
         return new ResponseEntity<>(
-                suggesterService.getSuggestions(getSuggestionDictionary(dict), query),
-                HttpStatus.OK);
+                Error.builder()
+                        .message(ex.getMessage())
+                        .code(httpStatus.value())
+                        .build(), httpStatus);
     }
+
+    @ExceptionHandler({SuggestionRetrievalException.class})
+    public ResponseEntity<Error> handleMissingParams(SuggestionRetrievalException ex) {
+        HttpStatus httpStatus = HttpStatus.INTERNAL_SERVER_ERROR;
+        return new ResponseEntity<>(
+                Error.builder()
+                        .message(ex.getMessage())
+                        .code(httpStatus.value())
+                        .build(), httpStatus);
+    }
+
+
+    @Data
+    @Builder
+    private static class Error {
```

Comment:
The error entity, does not have the same pattern that we use for the others. (url, messages)

Reply:
no prob now, since i've refactored as you suggested, to use our common response error handler